task6-hello-module-procfs/hello.c

```diff
@@ -0,0 +1,86 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/moduleparam.h>
+#include <linux/printk.h>
+#include <linux/proc_fs.h>
+#include <linux/uaccess.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Kriz");
+MODULE_DESCRIPTION("Hello_world module with simple calculator (sum, sub and mul)");
+MODULE_VERSION("0.1");
+
+#define PROC_BUFFER_SIZE 100
+
+static char   procfs_buffer[PROC_BUFFER_SIZE] = {0};
+static size_t procfs_buffer_size = 0; 
+
+static struct proc_dir_entry *proc_file;
+static struct proc_dir_entry *proc_folder;
+
+#define PROC_FILE_NAME "dummy"
+#define PROC_DIR_NAME  "hello"
+
+static ssize_t hello_read(struct file *File, char __user* buffer, size_t count, loff_t *offset)
+{
+	int ret;
+
+	if (*offset > 0)
+		ret = 0;
+
+	if (copy_to_user(buffer, procfs_buffer, procfs_buffer_size))
```

Comment:
In general here can be a case when calling user space application might provide buffer smaller then "procfs_buffer_size", so copy_to_user call will corrupt some other data.